airflow has 4 projects
[36m[[[airflow-5686]]][0m
Filename :  /home/wonseok/benchmark/airflow-5686/airflow/hooks/http_hook.py
[[[ Patch Node ]]]
if isinstance(conn.host, type(None)):
    conn.host = ''
[32mPASSED![0m
Time :  29.15 seconds

[36m[[[airflow-6036]]][0m
/home/wonseok/benchmark/airflow-6036/airflow/gcp/hooks/dataflow.py _get_dataflow_jobs
Filename :  /home/wonseok/benchmark/airflow-6036/airflow/gcp/hooks/dataflow.py
[[[ Patch Node ]]]
if isinstance(self._jobs, dict):
    self._jobs = [self._jobs]
[32mPASSED![0m
Time :  6.93 seconds

[36m[[[airflow-12094]]][0m
[31mFAILED...[0m
Time :  0.0 seconds

[36m[[[airflow-14513]]][0m
Filename :  /home/wonseok/benchmark/airflow-14513/airflow/kubernetes/pod_launcher.py
[[[ Patch Node ]]]
if not isinstance(last_log_time, type(None)):
    while True:
        logs = self.read_pod_logs(pod, timestamps=True, since_seconds=read_logs_since_sec)
        for line in logs:
            (timestamp, message) = self.parse_log_line(line.decode('utf-8'))
            last_log_time = pendulum.parse(timestamp)
            self.log.info(message)
        time.sleep(1)
        if not self.base_container_is_running(pod):
            break
        self.log.warning('Pod %s log read interrupted', pod.metadata.name)
        delta = pendulum.now() - last_log_time
        read_logs_since_sec = math.ceil(delta.total_seconds())
[32mPASSED![0m
Time :  36.69 seconds

PASSED :  3 / 4
beets has 1 projects
[36m[[[beets-3360]]][0m
Filename :  /home/wonseok/benchmark/beets-3360/beetsplug/thumbnails.py
[[[ Patch Node ]]]
if not isinstance(path, bytes):
    return PurePosixPath(path).as_uri()
[32mPASSED![0m
Time :  1.5 seconds

PASSED :  1 / 1
click has 1 projects
[36m[[[click-1551]]][0m
[31mFAILED...[0m
Time :  21.01 seconds

PASSED :  0 / 1
core has 8 projects
[36m[[[core-1972]]][0m
[31mFAILED...[0m
Time :  523.77 seconds

[36m[[[core-8065]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 161, in run
    asts, files_src = ast.get_asts(DIRECTORY, project)
  File "/home/wonseok/pyfix/my_tool/synthesizer/my_ast.py", line 32, in get_asts
    asts, files_src = self.files_to_asts(dir, project)
  File "/home/wonseok/pyfix/my_tool/synthesizer/my_ast.py", line 25, in files_to_asts
    tmp = ast.parse(src)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
invalid syntax (<unknown>, line 34)
Time :  523.8 seconds

[36m[[[core-20233]]][0m
Filename :  /home/wonseok/benchmark/core-20233/homeassistant/helpers/template.py
[[[ Patch Node ]]]
if isinstance(value, datetime.datetime):
    value = str(value)
[32mPASSED![0m
Time :  3.0 seconds

[36m[[[core-29829]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 81, in patch_only_once
    for (target_node, target_arg) in var_score.keys() :
cannot unpack non-iterable Name object
Time :  68.31 seconds

[36m[[[core-32222]]][0m
Filename :  /home/wonseok/benchmark/core-32222/homeassistant/components/unifi/device_tracker.py
[[[ Patch Node ]]]
if isinstance(self.client.last_seen, type(None)):
    return False
[32mPASSED![0m
Time :  6.18 seconds

[36m[[[core-32318]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 81, in patch_only_once
    for (target_node, target_arg) in var_score.keys() :
cannot unpack non-iterable Name object
Time :  103.74 seconds

[36m[[[core-40034-1]]][0m
None popleft
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 83, in patch_only_once
    chain_result = type_chain.get_type_chain(neg_file_node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 68, in get_type_chain
    candidate = find.do()
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 181, in do
    self.visit(stmt)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 162, in visit_Assign
    neg_type, pos_type = self.func_return_difference(call_filename, self.call_name)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 109, in func_return_difference
    if neg_func['path'] in call_filename and call_funcname in neg_func['func_name'] :
argument of type 'NoneType' is not iterable
Time :  1.36 seconds

[36m[[[core-40034-2]]][0m
None popleft
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 83, in patch_only_once
    chain_result = type_chain.get_type_chain(neg_file_node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 68, in get_type_chain
    candidate = find.do()
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 181, in do
    self.visit(stmt)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 162, in visit_Assign
    neg_type, pos_type = self.func_return_difference(call_filename, self.call_name)
  File "/home/wonseok/pyfix/my_tool/type_analysis/type_chain.py", line 109, in func_return_difference
    if neg_func['path'] in call_filename and call_funcname in neg_func['func_name'] :
argument of type 'NoneType' is not iterable
Time :  1.32 seconds

PASSED :  2 / 8
luigi has 1 projects
[36m[[[luigi-2323]]][0m
[31mFAILED...[0m
Time :  0.02 seconds

PASSED :  0 / 1
numpy has 3 projects
[36m[[[numpy-9999]]][0m
[31mFAILED...[0m
Time :  0.0 seconds

[36m[[[numpy-10473]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 81, in patch_only_once
    for (target_node, target_arg) in var_score.keys() :
cannot unpack non-iterable Name object
Time :  1.02 seconds

[36m[[[numpy-19094]]][0m
Filename :  /home/wonseok/benchmark/numpy-19094/numpy/typing/_generic_alias.py
[[[ Patch Node ]]]
if isinstance(i, typing._GenericAlias):
    continue
[32mPASSED![0m
Time :  4.16 seconds

PASSED :  1 / 3
pandas has 28 projects
[36m[[[pandas-15941]]][0m
[31mFAILED...[0m
Time :  10.94 seconds

[36m[[[pandas-17609]]][0m
Filename :  /home/wonseok/benchmark/pandas-17609/pandas/util/_decorators.py
[32mPASSED![0m
Time :  2.83 seconds

[36m[[[pandas-19276]]][0m
Filename :  /home/wonseok/benchmark/pandas-19276/pandas/core/indexes/datetimes.py
[[[ Patch Node ]]]
if isinstance(other, pandas._libs.tslibs.nattype.NaTType):
    return None
[32mPASSED![0m
Time :  5.66 seconds

[36m[[[pandas-20968-1]]][0m
Filename :  /home/wonseok/benchmark/pandas-20968-1/pandas/plotting/_core.py
[[[ Patch Node ]]]
if not isinstance(kwds, dict):
    for ((key, group), ax) in zip(grouped, axes):
        d = group.boxplot(ax=ax, column=column, fontsize=fontsize, rot=rot, grid=grid, **kwds)
        ax.set_title(pprint_thing(key))
        ret.loc[key] = d
[32mPASSED![0m
Time :  3157.64 seconds

[36m[[[pandas-20968-2]]][0m
Filename :  /home/wonseok/benchmark/pandas-20968-2/pandas/plotting/_core.py
[[[ Patch Node ]]]
if not isinstance(kwds, dict):
    for ((key, group), ax) in zip(grouped, axes):
        d = group.boxplot(ax=ax, column=column, fontsize=fontsize, rot=rot, grid=grid, **kwds)
        ax.set_title(pprint_thing(key))
        ret.loc[key] = d
[32mPASSED![0m
Time :  3318.14 seconds

[36m[[[pandas-21540]]][0m
Filename :  /home/wonseok/benchmark/pandas-21540/pandas/io/json/normalize.py
[32mPASSED![0m
Time :  108.84 seconds

[36m[[[pandas-21590-1]]][0m
[31mFAILED...[0m
Time :  204.89 seconds

[36m[[[pandas-21590-2]]][0m
[31mFAILED...[0m
Time :  174.73 seconds

[36m[[[pandas-22072]]][0m
[31mFAILED...[0m
Time :  781.64 seconds

[36m[[[pandas-22198]]][0m
Filename :  /home/wonseok/benchmark/pandas-22198/pandas/core/arrays/datetimes.py
[[[ Patch Node ]]]
if isinstance(other, numpy.datetime64):
    return None
[32mPASSED![0m
Time :  15.11 seconds

[36m[[[pandas-22378]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 122, in patch_only_once
    target_arg, target_node = entry.get_function_entry_info(info, neg_file_node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 38, in get_function_entry_info
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 31, in visit_FunctionDef
    self.generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 26, in visit_FunctionDef
    self.target_arg = args[self.info['index']+1].arg
list index out of range
Time :  0.3 seconds

[36m[[[pandas-22804]]][0m
Filename :  /home/wonseok/benchmark/pandas-22804/pandas/io/json/normalize.py
[[[ Patch Node ]]]
if isinstance(data, dict):
    data = [data]
[32mPASSED![0m
Time :  76.91 seconds

[36m[[[pandas-25533]]][0m
Filename :  /home/wonseok/benchmark/pandas-25533/pandas/core/series.py
[32mPASSED![0m
Time :  228.61 seconds

[36m[[[pandas-25759]]][0m
[31mFAILED...[0m
Time :  3126.32 seconds

[36m[[[pandas-26765]]][0m
Filename :  /home/wonseok/benchmark/pandas-26765/pandas/core/arrays/categorical.py
[32mPASSED![0m
Time :  7.09 seconds

[36m[[[pandas-28251]]][0m
[31mFAILED...[0m
Time :  763.67 seconds

[36m[[[pandas-30532]]][0m
[31mFAILED...[0m
Time :  337.17 seconds

[36m[[[pandas-32223]]][0m
Attribute(
    value=Compare(
        left=Name(id='casted', ctx=Load()),
        ops=[
            Eq()],
        comparators=[
            Name(id='values', ctx=Load())]),
    attr='all',
    ctx=Load())
[31mFAILED...[0m
Time :  2482.76 seconds

[36m[[[pandas-32953]]][0m
Filename :  /home/wonseok/benchmark/pandas-32953/pandas/core/reshape/concat.py
[[[ Patch Node ]]]
if issubclass(objs.__class__, collections.abc.Mapping):
    objs = dict(objs)
[32mPASSED![0m
Time :  5.33 seconds

[36m[[[pandas-33373]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 122, in patch_only_once
    target_arg, target_node = entry.get_function_entry_info(info, neg_file_node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 38, in get_function_entry_info
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 31, in visit_FunctionDef
    self.generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 25, in visit_FunctionDef
    if args[0].arg == 'self' :
list index out of range
Time :  1268.9 seconds

[36m[[[pandas-36950]]][0m
[31mFAILED...[0m
Time :  165.76 seconds

[36m[[[pandas-37736]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 122, in patch_only_once
    target_arg, target_node = entry.get_function_entry_info(info, neg_file_node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 38, in get_function_entry_info
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 31, in visit_FunctionDef
    self.generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 25, in visit_FunctionDef
    if args[0].arg == 'self' :
list index out of range
Time :  23.25 seconds

[36m[[[pandas-38220]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 73, in patch_only_once
    var_score = error_analysis.extract_score(error_stmt)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 274, in extract_score
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 106, in visit_Assign
    self.visit(node.value)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 132, in visit_Call
    super().generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 200, in visit_Attribute
    name = extract_value(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 187, in extract_value
    value_name = node.value.func.id
'Attribute' object has no attribute 'id'
Time :  168.89 seconds

[36m[[[pandas-38431]]][0m
Filename :  /home/wonseok/benchmark/pandas-38431/pandas/io/parsers.py
[[[ Patch Node ]]]
if isinstance(self.index_col, type(None)):
    return True
[32mPASSED![0m
Time :  18.95 seconds

[36m[[[pandas-39095]]][0m
Filename :  /home/wonseok/benchmark/pandas-39095/pandas/core/indexes/base.py
[[[ Patch Node ]]]
if isinstance(self, pandas.core.indexes.datetimes.DatetimeIndex):
    return indexer
[32mPASSED![0m
Time :  4142.6 seconds

[36m[[[pandas-41155]]][0m
[31mFAILED...[0m
Time :  1794.35 seconds

[36m[[[pandas-41406]]][0m
Filename :  /home/wonseok/benchmark/pandas-41406/pandas/core/frame.py
[[[ Patch Node ]]]
if issubclass(key.__class__, str):
    key = str(key)
[32mPASSED![0m
Time :  13.28 seconds

[36m[[[pandas-41915]]][0m
[31mFAILED...[0m
Time :  1343.85 seconds

PASSED :  13 / 28
rasa has 1 projects
[36m[[[rasa-8704]]][0m
[31mFAILED...[0m
Time :  0.0 seconds

PASSED :  0 / 1
requests has 4 projects
[36m[[[requests-3179]]][0m
Filename :  /home/wonseok/benchmark/requests-3179/requests/models.py
[[[ Patch Node ]]]
if self.content and (not self.encoding and len(self.content) > 3):
    encoding = guess_json_utf(self.content)
    if encoding is not None:
        try:
            return complexjson.loads(self.content.decode(encoding), **kwargs)
        except UnicodeDecodeError:
            pass
[32mPASSED![0m
Time :  31.69 seconds

[36m[[[requests-3368]]][0m
Filename :  /home/wonseok/benchmark/requests-3368/requests/models.py
[[[ Patch Node ]]]
if not isinstance(chunk_size, type(None)):
    raise TypeError('chunk_size must be an int, it is instead a %s.' % type(chunk_size))
[32mPASSED![0m
Time :  24.99 seconds

[36m[[[requests-3390]]][0m
None Type Casting Other Type
Type :  requests.auth.HTTPDigestAuth
[31mFAILED...[0m
Time :  1331.6 seconds

[36m[[[requests-4723]]][0m
Filename :  /home/wonseok/benchmark/requests-4723/requests/utils.py
[[[ Patch Node ]]]
if isinstance(parsed.hostname, type(None)):
    return True
[32mPASSED![0m
Time :  24.21 seconds

PASSED :  3 / 4
rich has 1 projects
[36m[[[rich-919]]][0m
[31mFAILED...[0m
Time :  414.69 seconds

PASSED :  0 / 1
salt has 11 projects
[36m[[[salt-33908]]][0m
Not Found
[36m[[[salt-38947]]][0m
Not Found
[36m[[[salt-40465]]][0m
Not Found
[36m[[[salt-50958]]][0m
Not Found
[36m[[[salt-52624]]][0m
Filename :  /home/wonseok/benchmark/salt-52624/salt/cli/batch.py
[32mPASSED![0m
Time :  3.54 seconds

[36m[[[salt-52980]]][0m
Not Found
[36m[[[salt-53394]]][0m
Filename :  /home/wonseok/benchmark/salt-53394/salt/utils/http.py
[[[ Patch Node ]]]
if isinstance(pgctnt, type(None)):
    return pgctnt
[32mPASSED![0m
Time :  4.68 seconds

[36m[[[salt-54240]]][0m
[31mFAILED...[0m
Time :  133.61 seconds

[36m[[[salt-54785]]][0m
[31mFAILED...[0m
Time :  400.76 seconds

[36m[[[salt-56094]]][0m
[31mFAILED...[0m
Time :  0.0 seconds

[36m[[[salt-56381]]][0m
[('__constant__', 'str')]
[(<ast.Subscript object at 0x7fcc1fa97af0>, 'typing.Any')]
[31mFAILED...[0m
Time :  2871.43 seconds

PASSED :  2 / 11
sanic has 2 projects
[36m[[[sanic-1334]]][0m
Filename :  /home/wonseok/benchmark/sanic-1334/sanic/blueprints.py
[[[ Patch Node ]]]
if isinstance(bp.url_prefix, type(None)):
    bp.url_prefix = ''
[32mPASSED![0m
Time :  6.68 seconds

[36m[[[sanic-2008]]][0m
[31mFAILED...[0m
Time :  0.0 seconds

PASSED :  1 / 2
scikitlearn has 3 projects
[36m[[[scikitlearn-7064]]][0m
Filename :  /home/wonseok/benchmark/scikitlearn-7064/sklearn/svm/base.py
[[[ Patch Node ]]]
if isinstance(kernel, bytes):
    return self
[32mPASSED![0m
Time :  50.72 seconds

[36m[[[scikitlearn-8973]]][0m
Filename :  /home/wonseok/benchmark/scikitlearn-8973/sklearn/linear_model/coordinate_descent.py
[[[ Patch Node ]]]
if isinstance(cv, sklearn.model_selection._split.StratifiedKFold):
    return self
[32mPASSED![0m
Time :  211.24 seconds

[36m[[[scikitlearn-17233]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 122, in patch_only_once
    target_arg, target_node = entry.get_function_entry_info(info, neg_file_node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 38, in get_function_entry_info
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 31, in visit_FunctionDef
    self.generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/synthesizer/get_function_entry_info.py", line 25, in visit_FunctionDef
    if args[0].arg == 'self' :
list index out of range
Time :  87.13 seconds

PASSED :  2 / 3
tornado has 1 projects
[36m[[[tornado-1689]]][0m
[31mFAILED...[0m
Time :  283.35 seconds

PASSED :  0 / 1
transformers has 1 projects
[36m[[[transformers-8052]]][0m
[33mERROR...[0m
  File "/home/wonseok/pyfix/my_tool/./test_main.py", line 170, in run
    works.work()
  File "/home/wonseok/pyfix/my_tool/work.py", line 169, in work
    self.patch_only_once(synthe)
  File "/home/wonseok/pyfix/my_tool/work.py", line 73, in patch_only_once
    var_score = error_analysis.extract_score(error_stmt)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 274, in extract_score
    self.visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 417, in generic_visit
    self.visit(value)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 132, in visit_Call
    super().generic_visit(node)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 415, in generic_visit
    self.visit(item)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 157, in visit_BinOp
    left = self.visit(node.left)
  File "/home/wonseok/.pyenv/versions/3.9.1/lib/python3.9/ast.py", line 407, in visit
    return visitor(node)
  File "/home/wonseok/pyfix/my_tool/type_analysis/error_analysis.py", line 164, in visit_BinOp
    left.extend(right)
'NoneType' object has no attribute 'extend'
Time :  0.53 seconds

PASSED :  0 / 1
Total :  28 / 70
